data-prepper-plugins/aggregate-processor/src/main/java/com/amazon/dataprepper/plugins/processor/aggregate/AggregateGroup.java

```diff
@@ -5,14 +5,45 @@
 
 package com.amazon.dataprepper.plugins.processor.aggregate;
 
+import java.time.Instant;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantLock;
+
 class AggregateGroup implements AggregateActionInput {
     private final GroupState groupState;
+    private Instant groupStart;
+    private final Lock concludeGroupLock;
+    private final Lock handleEventForGroupLock;
+
 
     AggregateGroup() {
         this.groupState = new DefaultGroupState();
+        this.groupStart = Instant.now();
+        this.concludeGroupLock = new ReentrantLock();
+        this.handleEventForGroupLock = new ReentrantLock();
     }
 
     public GroupState getGroupState() {
         return groupState;
     }
+
+    Instant getGroupStart() {
```

Comment:
Minor: Maybe specify the access level for these methods?

Reply:
They are meant to be package private, in order to keep them from being used in an `AggregateAction`. This previous PR contains a description that may make this more clear (https://github.com/opensearch-project/data-prepper/pull/945)

Reply:
@dapowers87 , This is the way to express package private members in Java. There is no keyword for it.